lib/mysqlDao.js

```diff
@@ -104,6 +104,20 @@ class MysqlDao {
             });
     }
 
+    upsertBulk(insertArgs, onDupUpdateArgs) {
```

Comment:
I would add a JSDoc comment describing the parameters, especially noting that onDupUpdateArgs can be either an array (and will use `col = VALUES(col)`) or an object (in which case standard logic applies like `col = ${col}`